src/main/java/org/intellij/sdk/codesync/CodeSyncLogger.java

```diff
@@ -0,0 +1,119 @@
+            String sequenceToken = describeLogStreamsResponse.logStreams().get(0).uploadSequenceToken();
+
+            // Build an input log message to put to CloudWatch.
+            JSONObject msg = new JSONObject();
+            msg.put("msg", message);
+            msg.put("source", DIFF_SOURCE);
+            InputLogEvent inputLogEvent = InputLogEvent.builder()
+                    .message(msg.toJSONString())
+                    .timestamp(System.currentTimeMillis())
+                    .build();
+
+            // Specify the request parameters.
+            // Sequence token is required so that the log can be written to the
+            // latest location in the stream.
+            PutLogEventsRequest putLogEventsRequest = PutLogEventsRequest.builder()
+                    .logEvents(Collections.singletonList(inputLogEvent))
+                    .logGroupName(logGroupName)
+                    .logStreamName(streamName)
+                    .sequenceToken(sequenceToken)
+                    .build();
+
+            logsClient.putLogEvents(putLogEventsRequest);
+
+            System.out.println("Successfully put CloudWatch log event");
+        } catch (CloudWatchException e) {
+            e.printStackTrace();
+            throw e;
+        }
+    }
+
+    public static void logEvent(String message, String userEmail) {
+        LOGGER.log(Level.SEVERE, message);
+        UserFile.User user = null;
+        UserFile userFile = null;
+
+        try {
+            userFile = new UserFile(USER_FILE_PATH);
+        } catch (FileNotFoundException | InvalidYmlFileError e) {
+            e.printStackTrace();
+        }
+
+        if (userEmail != null && userFile != null) {
+            user = userFile.getUser(userEmail);
+        } else if (userFile != null) {
+            user = userFile.getUser();
+        }
+
+//        SequenceTokenFile sequenceTokenFile;
+        if (user == null) {
+            // Can't log any thing.
+            return;
+        } else if (user.getAccessKey() == null || user.getSecretKey() == null || user.getUserEmail() == null) {
+            // can't log anything
+            return;
+        }
+        // I do not think we need a separate sequence token file in java. Need to verify it though.
+//        else {
+//            try {
+//                sequenceTokenFile = new SequenceTokenFile(SEQUENCE_TOKEN_FILE_PATH);
+//            } catch (FileNotFoundException | InvalidYmlFileError e) {
+//                // Can't log anything;
+//                return;
+//            }
+//        }
+//        SequenceTokenFile.SequenceToken sequenceToken = sequenceTokenFile.getSequenceToken(user.getUserEmail());
+
+        try {
+            logMessageToCloudWatch(
+                    CLIENT_LOGS_GROUP_NAME, user.getUserEmail(), user.getAccessKey(), user.getSecretKey(), message
+            );
+        } catch (CloudWatchException e) {
+            // TODO: Shouldn't we handle infinite loop here?
```

Comment:
I think yes, but we decided to keep it infinte loop for now

Reply:
I limited the iterations to 10 max, is that okay?

Reply:
Let's hear @osman-masood's opinion as well. I think it shouldn't be infinite. OR maybe if there are similar infinte logs, send log to AWS only once every X minutes maybe. i.e. reduce calls to AWS

Reply:
Oh, I see that you have put limit on the retry part. not to sending logs to AWS. That's fine I guess. Adding in sheet for other plugins as well